### PR TITLE
fix(#42): properly escape double-quotes in string fields

### PR DESCRIPTION
**Pull request by orzechowskid**
plus some refactoring to make unit-testing easier

### CODE REVIEW

File: src/generate.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export const csvGenerateRow = (
+  row: any,
+  headerKeys: string[],
+  delimiter: string
+) =>
+  headerKeys.map(
+    (fieldName) => typeof row[fieldName] === 'number'
+      ? row[fieldName]
+      : `"${String(row[fieldName]).replace(/"/g, '""')}"`
```
Comment (orzechowskid) [inline src/generate.ts:9]: numbers are inserted without any processing.  other datatypes are cast to string, have any double-quote characters escaped by adding a second double-quote character, and then the entire string is wrapped in double-quotes similar to how `JSON.stringify()` does.

File: src/index.ts [hunk 2]
```diff
@@ -22,18 +27,8 @@ const csvDownload = ({
...
-  const columnNames = headers ?? headerKeys;
-  const csv = data.map((row) =>
-    headerKeys
-      .map((fieldName) =>
-        JSON.stringify(row[fieldName] === 0 ? 0 : row[fieldName] ?? "")
-      )
-      .join(delimiter)
-  );
-  csv.unshift(columnNames.join(delimiter));
-  const csvAsString = csv.join("\r\n");
```
Comment (orzechowskid) [inline src/index.ts:36]: I moved this to a separate file so that it could be tested.

File: test/generate.spec.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  const mockData = {
+    id: 1,
+    text: 'Lee "Scratch" Perry'
+  };
+
+  expect(csvGenerateRow(mockData, [ 'id', 'text' ], ';'))
+    .toEqual(
+      `1;"Lee ""Scratch"" Perry"`
+    );
+});
```
Comment (orzechowskid) [inline test/generate.spec.ts:27]: happy to add any other test cases for the csv-building logic that you think might be useful.

File: src/generate.ts (hunk 1 above)
Comment (coston) [reply src/generate.ts:9]: Nice!

File: src/index.ts (hunk 2 above)
Comment (coston) [reply src/index.ts:36]: Good call.

File: test/generate.spec.ts (hunk 3 above)
Comment (coston) [reply test/generate.spec.ts:27]: This is very helpful! Thank you.

**Review by coston — APPROVED** [review]
Looks good! Nice improvement.

**Pull request merged**